pandas/core/internals/managers.py

```diff
@@ -938,13 +1035,24 @@ def _verify_integrity(self) -> None:
                 f"block items\n# manager items: {len(self.items)}, # "
                 f"tot_items: {tot_items}"
             )
+        if self.refs is not None:
+            if len(self.refs) != len(self.blocks):
+                raise ValueError(
```

Comment:
Would this exception be internally caught? IMO probably not entirely clear what the action would be from a user if they encountered this exception

Reply:
Or maybe elsewhere we use AssertionError for internal errors?

Reply:
> if you see this error, please open a bug report

Yeah that would be helpful to include here. Makes sense that this should be internal